Feature/wdl1.0

aofarrel opened the pull request:
Converts the aligner workflow to match WDL 1.0 specifications.

Overall changes from this versus draft WDL:

- Indentation is more consistent (not part of the specs, but makes it prettier)
- Inputs are {encased in braces} as required under WDL 1.0
- WDL 1.0 introduces a seemingly undocumented change in how float/ints are handled, so a workaround was devised using strings
- Usage of $ versus ~ for variables in a task with
```
command <<<
   foo
>>> 
```
syntax, as opposed to the more problematic
```
command {
    foo
}
```
syntax, now complies to WDL 1.0 standards. While it would pass womtool just fine either way, without these changes it will fail upon execution due to it incorrectly thinking some variables were unbound. This affects the `dollar="$"` workaround that already existed in the WDL but does not make the workaround unnecessary. (Oddly enough, seemingly arbitrary variables work fine in $ notation, but for consistency’s sake I have tried to make it completely match the specification.)

This has been tested on my local machine.

Commit:
Merge tag '1.31.0' into develop

Use https paths instead of GCS paths in input CWL JSON

Ldcabansay committed:
Add metadata to workflows published on dockstore (#89)

* add metadata to u_of_michigan_aligner.wdl

* add metadata to FunctionalEquivalence.wdl

* add metadata for topmed_freeze3_calling.wdl

* add metadata sbg-alignment-cwl/topmed-alignment.cwl  

* add metadata functional equivalent aligner: /topmed-cwl/workflow/alignment_workflow.cwl

* add metadata to topmed_freeze3_calling.cwl

* add metadata topmed_variant_calling_pipeline.cwl

Ldcabansay committed:
update raw imports for 1.32.0 release (#90)

aofarrel committed:
CRAIs now optional and gs style URLs in add'l JSON

aofarrel committed:
Merge pull request #101 from DataBiosphere/feature/CRAI_GS

CRAIs now optional and gs style URLs in add'l JSON

aofarrel committed:
made CRAI files optional and added JSON with gs:// URIs

aofarrel committed:
added input {}s and fixed some spacing

aofarrel committed:
(do not use) Float/Int fixes, better indentation

align steps fails due to unbound variable; likely an issue in the handoff of the output of the prealign step

See comments for the float/int workarounds needed

aofarrel committed:
Update u_of_michigan_aligner.wdl

aofarrel committed:
Update u_of_michigan_aligner.wdl

aofarrel committed:
First fully-functional version

Confirmed to run locally

aofarrel committed:
removed comments, fixed spacing

aofarrel committed:
Merge pull request #104 from DataBiosphere/feature-wdl1.0

Feature wdl1.0

wshands commented on aligner/u_of_michigan_aligner/u_of_michigan_aligner.wdl:
```diff
@@ -11,152 +12,167 @@
+    # must use the reference genome that the CRAM was originally aligned with
+    # to convert the CRAM to a SAM
+    File? PreAlign_reference_genome
+    File PreAlign_reference_genome_default = select_first([PreAlign_reference_genome,ref_fasta])
```
I think with WDL 1.0 you can simplify a default setting for a variable like this:
```
File PreAlign_reference_genome_default = ref_fasta
```
I think in this case if PreAlign_reference_genome_default is not provided in the input json it uses ref_fasta; but you will need to verify this. See https://github.com/DataBiosphere/topmed-workflow-variant-calling/blob/master/variant-caller-wdl/topmed_freeze8_caller.wdl

https://github.com/openwdl/wdl/blob/master/versions/1.0/SPEC.md#declared-inputs-defaults-and-overrides

aofarrel replied:
In my opinion it's better to leave it as-is because the logic of the new option is harder to follow -- ie, it's not immediately obvious that it is a default that can be overwritten, while with `select_first()` both situations are immediately clear.

wshands commented on aligner/u_of_michigan_aligner/u_of_michigan_aligner.wdl:
```diff
@@ -11,152 +12,167 @@
+
+    String PreAlign_disk_size_string = PreAlign_disk_size
+    String PreAlign_disk_size_before_decimal = sub(PreAlign_disk_size_string, "\\..*", "")
+    Int PreAlign_disk_size_int = PreAlign_disk_size_before_decimal
```
You can convert from a Float to an Int by using ceil if you want, as in  disks: "local-disk " + ceil(disk_size) + " HDD"
https://github.com/DataBiosphere/topmed-workflow-variant-calling/blob/2bc37748e11d04f5d5a3ee69029947fce7c55aee/variant-caller-wdl/topmed_freeze8_caller.wdl#L543

aofarrel committed:
hotfix for syntax error in comment

aofarrel committed:
Removed the "string hack"

Runtime attribute disk_size requires type int in its string. Previously we kind of cheated by using strings to coerce a float into an int. With the built-in ceil() we can now make things a lot simplier. Ceil() will of course round up, but that was the end result of the string hack anyway.

wshands commented:
You might want to figure out how to fix up the TravisCI tests to make them work...

aofarrel committed:
wdl 1.0-like checker changes

untested, this push is needed to test on Terra

aofarrel committed:
wdl 1.0-like formatting for checker

continuation of previous commit

aofarrel committed:
remove sub() workaround, which doesn't work in wdl 1.0

aofarrel committed:
attempt to fix disk_size issue

aofarrel committed:
test if issue is additional_disk

aofarrel committed:
cleanup

Checked to work on Terra

aofarrel commented:
I've managed to fix the checker and update both of its files to WDL 1.0, and this passed a test on Terra. 🎉 

There will be one more push directly to master after merging which will update the import links in the checker WDL, which currently link to this branch, and instead having them import from master (which by then will have the necessary changes reflected), allowing this branch to be safely deleted.

aofarrel committed:
Merge branch 'master' into feature/wdl1.0

aofarrel commented:
Branch now to be deleted after [this commit](https://github.com/DataBiosphere/topmed-workflows/commit/de683e0be7c2798b3eb503812a095659039247b3).
